From 38ac6abaa528e5ade42d60155aea7e1ea4e80899 Mon Sep 17 00:00:00 2001
From: Junqiu Lei <junqiu@amazon.com>
Date: Fri, 1 Aug 2025 17:20:56 -0700
Subject: [PATCH] Custom patch to support range search params

Signed-off-by: Junqiu Lei <junqiu@amazon.com>
---
 faiss/IndexIDMap.cpp | 28 ++++++++++++++++++++++++----
 1 file changed, 24 insertions(+), 4 deletions(-)

diff --git a/faiss/IndexIDMap.cpp b/faiss/IndexIDMap.cpp
index 123e8f2ff..8c7a893e3 100644
--- a/faiss/IndexIDMap.cpp
+++ b/faiss/IndexIDMap.cpp
@@ -256,11 +256,31 @@ void IndexIDMapTemplate<IndexT>::range_search(
         RangeSearchResult* result,
         const SearchParameters* params) const {
     if (params) {
-        SearchParameters internal_search_parameters;
-        IDSelectorTranslated id_selector_translated(id_map, params->sel);
-        internal_search_parameters.sel = &id_selector_translated;
+        IDSelectorTranslated this_idtrans(this->id_map, nullptr);
+        ScopedSelChange sel_change;
+        IDGrouperTranslated this_idgrptrans(this->id_map, nullptr);
+        ScopedGrpChange grp_change;
+
+        if (params->sel) {
+            auto idtrans = dynamic_cast<const IDSelectorTranslated*>(params->sel);
+
+            if (!idtrans) {
+                auto params_non_const = const_cast<SearchParameters*>(params);
+                this_idtrans.sel = params->sel;
+                sel_change.set(params_non_const, &this_idtrans);
+            }
+        }
+
+        if (params->grp) {
+            auto idtrans = dynamic_cast<const IDGrouperTranslated*>(params->grp);
+            if (!idtrans) {
+                auto params_non_const = const_cast<SearchParameters*>(params);
+                this_idgrptrans.grp = params->grp;
+                grp_change.set(params_non_const, &this_idgrptrans);
+            }
+        }
 
-        index->range_search(n, x, radius, result, &internal_search_parameters);
+        index->range_search(n, x, radius, result, params);
     } else {
         index->range_search(n, x, radius, result);
     }
-- 
2.39.5 (Apple Git-154)
