From 38d32faf4183ad400831624a3d6874af6128f9a8 Mon Sep 17 00:00:00 2001
From: Owen Halpert <ohalpert@amazon.com>
Date: Mon, 31 Mar 2025 17:00:48 -0700
Subject: [PATCH] Bump cmake version nmslib

---
 similarity_search/CMakeLists.txt | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/similarity_search/CMakeLists.txt b/similarity_search/CMakeLists.txt
index bc6ef3c..c555115 100644
--- a/similarity_search/CMakeLists.txt
+++ b/similarity_search/CMakeLists.txt
@@ -8,7 +8,11 @@
 #
 #
 
-cmake_minimum_required (VERSION 2.8)
+cmake_minimum_required (VERSION 3.5...4.0)
+
+if(CMAKE_VERSION VERSION_GREATER_EQUAL "4.0")
+    cmake_policy(SET CMP0153 OLD)
+endif()
 
 project (NonMetricSpaceLib)
 
@@ -20,12 +24,10 @@ project (NonMetricSpaceLib)
 #
 function(CXX_COMPILER_DUMPVERSION _OUTPUT_VERSION)
 
-  exec_program(${CMAKE_CXX_COMPILER}
-    ARGS ${CMAKE_CXX_COMPILER_ARG1} -dumpversion
-    OUTPUT_VARIABLE COMPILER_VERSION
+  execute_process(
+      COMMAND ${CMAKE_CXX_COMPILER} ${CMAKE_CXX_COMPILER_ARG1} -dumpversion
+      OUTPUT_VARIABLE COMPILER_VERSION
   )
-  #string(REGEX REPLACE "([0-9])\\.([0-9])(\\.[0-9])?" "\\1\\2"
-  #   COMPILER_VERSION ${COMPILER_VERSION})
 
   set(${_OUTPUT_VERSION} ${COMPILER_VERSION} PARENT_SCOPE)
 endfunction()
@@ -55,7 +57,7 @@ elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Intel")
     endif()
     set (CMAKE_CXX_FLAGS_RELEASE "-Wall -Wunreachable-code -Ofast -DNDEBUG -std=c++11 -DHAVE_CXX0X -pthread ${SIMD_FLAGS} -fpic")
     set (CMAKE_CXX_FLAGS_DEBUG   "-Wall -Wunreachable-code -ggdb  -DNDEBUG -std=c++11 -DHAVE_CXX0X -pthread ${SIMD_FLAGS} -fpic")
-elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang")
+elseif(${CMAKE_CXX_COMPILER_ID} STREQUAL "Clang" OR ${CMAKE_CXX_COMPILER_ID} STREQUAL "AppleClang")
     if (CMAKE_SYSTEM_NAME MATCHES Darwin)
         # MACOSX
         set (CMAKE_CXX_FLAGS_RELEASE "${WARN_FLAGS} -O3 -DNDEBUG -std=c++11 -DHAVE_CXX0X -pthread -fpic ${SIMD_FLAGS}")
-- 
2.47.1

